src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.tsx

```diff
@@ -1,64 +1,141 @@
-import { getGenomeInfo } from 'src/genome/genomeSelectors';
+import { updateTrackStatesAndSave } from 'src/content/app/browser/browserActions';
+import { BrowserTrackStates } from 'src/content/app/browser/track-panel/trackPanelConfig';
+import { getActiveGenomePreviouslyViewedObjects } from 'src/content/app/browser/track-panel/trackPanelSelectors';
 import { fetchExampleEnsObjects } from 'src/ens-object/ensObjectActions';
 import { getExampleEnsObjects } from 'src/ens-object/ensObjectSelectors';
 import * as urlFor from 'src/shared/helpers/urlHelper';
-import { getFormattedLocation } from 'src/shared/helpers/regionFormatter';
-
-import upperFirst from 'lodash/upperFirst';
+import { closeTrackPanelModal } from '../../trackPanelActions';
+import ImageButton from 'src/shared/components/image-button/ImageButton';
+import { ReactComponent as EllipsisIcon } from 'static/img/track-panel/ellipsis.svg';
+import { changeDrawerViewAndOpen } from 'src/content/app/browser/drawer/drawerActions';
+import { PreviouslyViewedObject } from 'src/content/app/browser/track-panel/trackPanelState';
 
 import styles from '../TrackPanelModal.scss';
+import { Status } from 'src/shared/types/status';
 
-type TrackPanelBookmarksProps = {
+export type TrackPanelBookmarksProps = {
   activeGenomeId: string | null;
-  genomeInfo: GenomeInfoData;
   exampleEnsObjects: EnsObject[];
+  previouslyViewedObjects: PreviouslyViewedObject[];
   fetchExampleEnsObjects: (objectId: string) => void;
+  updateTrackStatesAndSave: (trackStates: BrowserTrackStates) => void;
+  closeTrackPanelModal: () => void;
+  changeDrawerViewAndOpen: (drawerView: string) => void;
 };
 
-export const TrackPanelBookmarks = (props: TrackPanelBookmarksProps) => {
-  const getExampleObjLabel = (exampleObject: EnsObject) => {
-    if (exampleObject.object_type === 'gene') {
-      return exampleObject.label;
-    } else {
-      return getFormattedLocation(exampleObject.location);
-    }
-  };
+type ExampleLinksProps = Pick<
+  TrackPanelBookmarksProps,
+  'exampleEnsObjects' | 'activeGenomeId' | 'closeTrackPanelModal'
+>;
+export const ExampleLinks = (props: ExampleLinksProps) => {
+  return (
+    <div>
+      {props.exampleEnsObjects.map((exampleObject) => {
+        const path = urlFor.browser({
+          genomeId: props.activeGenomeId,
+          focus: exampleObject.object_id
+        });
 
-  const getPreviouslyViewed = () => {
-    return props.exampleEnsObjects.map((exampleObject) => {
-      const path = urlFor.browser({
-        genomeId: props.activeGenomeId,
-        focus: exampleObject.object_id
-      });
+        return (
+          <dd key={exampleObject.object_id}>
+            <Link to={path} onClick={props.closeTrackPanelModal}>
+              {exampleObject.label}
+            </Link>
+            <span className={styles.previouslyViewedType}>
+              {upperFirst(exampleObject.object_type)}
+            </span>
+          </dd>
```

Comment:
can we slowly get rid of dt's and dd's when do not have definition-related semantics?

Reply:
That's done